lib/hawkular/metrics/metric_api.rb

```diff
@@ -36,7 +36,22 @@ def push_data(gauges: [], counters: [], availabilities: [])
       availabilities.each { |g| default_timestamp g[:data] }
 
       data = { gauges: gauges, counters: counters, availabilities: availabilities }
-      http_post('/metrics/data', data)
+      path = 'metrics/'
+      if new_api_supported
+        path << 'raw'
+      else
+        path << 'data'
+      end
+      http_post(path, data)
+    end
+
+    def new_api_supported
```

Comment:
yes, I think private section might be better. 
also "new api" is not the best terminology, as then there might be later also "new new api" etc. 
versus just "api" (=current api) and legacy api.